File: data/readable_format/cpp/file1830.cpp
#include <vector>
#include <iostream>
#include <gtest/gtest.h>

using DOUBLE = double; // Assuming DOUBLE is equivalent to 'double'
const int nmeci = 2; // Example value for nmeci, adjust as necessary for your use case

class Deri2_I {
public:
    static void deri2(int minear, std::vector<std::vector<DOUBLE>>& f, const std::vector<std::vector<DOUBLE>>& fd,
                      std::vector<std::vector<DOUBLE>>& fci, int ninear, int nvar, 
                      std::vector<DOUBLE>& grad, DOUBLE throld, const std::vector<DOUBLE>& diag,
                      const std::vector<DOUBLE>& scalar) {
        // Simplified logic example for demonstration purposes
        for (int i = 0; i < nvar; ++i) {
            grad[i] = 0.0;
            for (int j = 0; j < minear; ++j) {
                grad[i] += f[j][i];
            }
            grad[i] /= static_cast<DOUBLE>(minear);
        }
    }
};

// Unit test for the Deri2_I class
TEST(Deri2Test, BasicTest) {
    int minear = 2;
    int ninear = 2;
    int nvar = 3;
    std::vector<std::vector<DOUBLE>> f = {{1.0, 2.0, 3.0}, {4.0, 5.0, 6.0}};
    std::vector<std::vector<DOUBLE>> fd(ninear, std::vector<DOUBLE>(nvar, 0.0));
    std::vector<std::vector<DOUBLE>> fci(ninear, std::vector<DOUBLE>(10, 0.0));
    std::vector<DOUBLE> grad(nvar, 0.0);
    std::vector<DOUBLE> diag(1, 0.0);
    std::vector<DOUBLE> scalar(1, 0.0);

    Deri2_I::deri2(minear, f, fd, fci, ninear, nvar, grad, 1.0, diag, scalar);

    // Check results
    EXPECT_DOUBLE_EQ(grad[0], 2.5);
    EXPECT_DOUBLE_EQ(grad[1], 3.5);
    EXPECT_DOUBLE_EQ(grad[2], 4.5);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/readable_format/fortran/file1830.f90
PROGRAM Main
  IMPLICIT NONE

  ! Assuming nmeci's value for demonstration purposes. Replace with the actual value.
  INTEGER, PARAMETER :: nmeci = 2
  INTEGER, PARAMETER :: nvar = 3, minear = 2, ninear = 2
  REAL(8), DIMENSION(minear,nvar) :: F
  REAL(8), DIMENSION(ninear,nvar) :: FD
  REAL(8), DIMENSION(ninear,nmeci*(nmeci + 1)*10/ninear) :: FCI
  REAL(8), DIMENSION(nvar) :: GRAD
  REAL(8) :: THROLD
  REAL(8), DIMENSION(nvar) :: DIAG
  REAL(8), DIMENSION(1) :: SCALAR

  ! Initialize variables
  F = RESHAPE([1.0d0, 2.0d0, 3.0d0, 4.0d0, 5.0d0, 6.0d0], SHAPE(F))
  FD = 0.0
  FCI = 0.0
  GRAD = 0.0
  THROLD = 1.0d0
  DIAG = 0.0
  SCALAR = 0.0

  ! Call the subroutine
  CALL deri2(minear, F, FD, FCI, ninear, nvar, GRAD, THROLD, DIAG, SCALAR)

  ! Output for demonstration
  PRINT *, 'GRAD:', GRAD

CONTAINS
  SUBROUTINE deri2 (MINEAR, F, FD, FCI, NINEAR, NVAR, GRAD, THROLD, DIAG, SCALAR)
    INTEGER, INTENT(IN) :: NINEAR, MINEAR, NVAR
    REAL(8), DIMENSION(MINEAR,NVAR), INTENT(INOUT) :: F
    REAL(8), DIMENSION(NINEAR,NVAR) :: FD
    REAL(8), DIMENSION(NINEAR,NMECI*(NMECI + 1)*10/NINEAR) :: FCI
    REAL(8), DIMENSION(NVAR), INTENT(INOUT) :: GRAD
    REAL(8), INTENT(IN) :: THROLD
    REAL(8), DIMENSION(*) :: DIAG
    REAL(8), DIMENSION(*), INTENT(IN) :: SCALAR

    INTEGER :: i, j
    ! Simplified logic for demonstration
    DO i = 1, NVAR
      GRAD(i) = 0.0d0
      DO j = 1, MINEAR
        GRAD(i) = GRAD(i) + F(j, i)
      END DO
      GRAD(i) = GRAD(i) / MINEAR
    END DO
  END SUBROUTINE deri2

END PROGRAM Main